common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -24,6 +26,34 @@ class EntitlementViewSet(viewsets.ModelViewSet):
     filter_backends = (DjangoFilterBackend,)
     filter_class = CourseEntitlementFilter
 
+    def retrieve(self, request, *args, **kwargs):
+        """
+        Override the retrieve method to expire a record that is past the
+        policy and is requested via the API before returning that record.
+        """
+        entitlement = self.get_object()
+        entitlement.expired_at_datetime  # Has an effect, updates the expired_at if it is expired
+        serializer = self.get_serializer(entitlement)
+        return Response(serializer.data)
+
+    def list(self, request, *args, **kwargs):
+        """
+        Override the list method to expire records that are past the
+        policy and requested via the API before returning those records.
+        """
+        queryset = self.filter_queryset(self.get_queryset())
+        with transaction.atomic():
+            for entitlement in queryset:
+                entitlement.expired_at_datetime  # Has an effect, updates the expired_at if it is expired
```

Comment:
Nit: For readability, it might be good to move whatever logic updates the `expired_at` time in `expired_at_datetime` into its own method (maybe `set_expired_at`) and call that here instead. It feels a little weird to call properties soly for their side effects.

Reply:
Agreed, pulling that into it's own method that will be referenced here, will do away with the wonky comment too :+1: